src/datastructure.h

```diff
@@ -57,6 +60,10 @@ typedef struct {
 	int overTime[OVERTIME_SLOTS];
 	unsigned int numQueriesARP;
 	bool new;
+	sqlite3_stmt* whitelist_stmt;
+	sqlite3_stmt* gravity_stmt;
+	sqlite3_stmt* blacklist_stmt;
```

Comment:
I think the size of these objects should always be known (`sizeof sqlite3_stmt*`), however, I agree that they are something the API will not be able to access anyway, so I'll move is into a separate FTL-only struct.

Reply:
Checking this, it would require a lot of extra housekeeping. As they are just pointers, they should always be 8 byte on 64bit targets and 4 byte on 32bit targets. The new `regex_enabled` is worth two pointers so either 16 or 8 bytes.

*edit*: Confirmed at least for x86_64:
```
(gdb) p sizeof(sqlite3_stmt*)
$1 = 8
(gdb) p sizeof(client->whitelist_stmt)
$2 = 8

(gdb) p sizeof(bool*[2])
$3 = 16
(gdb) p sizeof(client->regex_enabled)
$4 = 16
```